libraries/TuneScope/TuneScope.js

```diff
@@ -26,6 +26,13 @@ SnapExtensions.primitives.set(
     }
 );
 
+SnapExtensions.primitives.set(
+    'ts_getcurrentnote()',
+    function () {
+        return window.currentNote
```

Comment:
I'm puzzled by this. What happens if you play a chord? Or polyphonic music? Which note is going to be the "current" one? Seems like there can only be one *global* "current" note and whichever gets triggered last "wins" regardless of their duration. Frankly, this doesn't make sense to me. In a polyphonic / harmonic setting asking for a "current note" doesn't make sense at all. I would leave it up to the user script to keep track of this. I'm curious why you need a "current note" and what use cases you are envisioning.

Reply:
@jmoenig We completely agree about abuse of the global name space.  …  We need to synchronize animation on the Snap_!_ stage with sequences of musical notes. We had a “Current Note” block that provided the status of notes in the first track. This allowed us to trigger events on the stage with the timing of the music. We used the first track as the default for synchronization because that seemed as good a choice as any.

The students have seemed content to use melody tracks rather than chord tracks for this purpose. If there were a use case, this variable could conceivably provide either the note [C4] or the list [ [C4] [E4] [G4] ] in the case of a chord.  

We had some discussions about potential stripping out the list of durations in the first track, and using these to time the durations of the notes asynchronously. Latencies in delivery of the notes could cause this to drift, but we could look into it if this were judged desirable in order to avoid namespace problems.

Possibly there is a more elegant way to determine the timing of the notes and synchronize them with events on the Snap! stage, but so far we have not thought of another way to do this.